### PR TITLE
[SANS984] add dev notes to explain EventList::filterInPlaceHelper

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

- Original Gitlab Task: [SANS984](https://code.ornl.gov/sns-hfir-scse/sans/sans-backend/-/issues/984)

tl;dr;
-----

Added developer notes as comments to explain the two pointer technique used in `EventList::filterInPlaceHelper`.

Details
--------

As part of the efforts in developing TimeROI, a refactor effort was proposed for `EventList` in preparation of the upcoming replacement of TimeSplitter.
The two pointer scanning technique used in `EventList::filterInPlaceHelper` was not clearly understood and was labeled for refactor process.
Upon close inspection, the current implementation is sufficiently optimized for TimeROI and it was decided that the refactor effort should be changed into a documentation update.

**To test:**

No testing is required as this is a in-code developer documentation improvement.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by peterfpeterson — COMMENTED** [review]
Is there a reason why you didn't put this in the doxygen block (i.e. L3774)? I don't know that "users" ever really look at doxygen or comments in code. Looking at either of those implies developer notes. I to appreciate the content. The text is quite helpful.

**Review by peterfpeterson — COMMENTED** [review]
I think the `@brief` should be on L3772

**Review by peterfpeterson — APPROVED** [review]
Useful additional comment

**Comment by martyngigg** [comment]
https://github.com/mantidproject/mantid/pull/34930 clears the macOS compiler warning that was already there and not caused by this.
